### PR TITLE
Change Project to use new Elasticsearch method

### PR DESCRIPTION
**Pull request by ms10398**
- Full Text Search for projects
- Regex search for projects

Its WIP for now and I will be adding more commits to it

### CODE REVIEW

File: src/pages/project/project.pug [hunk 1]
```diff
@@ -46,15 +46,27 @@ div
...
-    //- template(slot="meta")
-    //-   div.row.right-side-slot
+  u-layout-page.u-page-project-tabs(reverse)
+    template(slot="meta")
+      div.row.right-side-slot
+        div.top-contributors
+          span.q-title.uppercase top contributors
+          q-list(separator).row
+            q-item(v-for="contributor in contributors" :key="contributor.key").col-12
+              q-item-side(:avatar="'https://steemitimages.com/u/'+contributor.key+'/avatar'")
```
Comment (gregory-latinier) [inline src/pages/project/project.pug]: Does template notation work on vue files ? If yes could you update it?
`https://steemitimages.com/u/${contributor.key}/avatar`

File: src/pages/project/project.vue [hunk 2]
```diff
@@ -37,6 +32,15 @@ export default {
       this.project = querySnapshot.docs[0].data()
       this.loading = false
     },
+    async loadContributors () {
+      const call = firebase.functions().httpsCallable(`/api/projects/contributors?q=${this.$route.params.name}`)
+      await call()
```
Comment (gregory-latinier) [inline src/pages/project/project.vue]: No need to set a variable you can just chain functions

File: src/pages/project/project.vue [hunk 3]
```diff
@@ -9,20 +10,14 @@ export default {
...
-      loading: false
-      // contributors: [
-      //   { id: 1, name: 'icaro', numberOfContributions: 17 },
-      //   { id: 2, name: 'hernandev', numberOfContributions: 23 },
-      //   { id: 3, name: 'elear', numberOfContributions: 12 },
-      //   { id: 4, name: 'espoem', numberOfContributions: 91 },
-      //   { id: 5, name: 'mkt', numberOfContributions: 21 }
-      // ],
+      loading: false,
       // taskRequests: [
```
Comment (gregory-latinier) [inline src/pages/project/project.vue]: Can remove this commented code?

File: src/pages/project-search/project-search.vue [hunk 4]
```diff
@@ -73,7 +73,7 @@ export default {
     // load posts main method.
     async loadProjects (done) {
       const listProjects = firebase.functions().httpsCallable(`api/projects/list?q=${this.search}&openSource=${this.opensource}`)
-      this.projects = (await listProjects()).data.map(project => project.data)
+      this.projects = (await listProjects()).data
```
Comment (gregory-latinier) [inline src/pages/project-search/project-search.vue]: Could you simplify the code at line 69?
`index and done` will always be undefined in this situation

```
loadProjectsScroll: debounce(() => this.loadProjects(), 3000),
```

and then in loadProjects remove this done things with the attempt.
Most of the time from what I've seen `done` is undefined so attempt does nothing it's just overload the code.
thank you

File: src/pages/project/project.vue (hunk 3 above)
Comment (ms10398) [reply src/pages/project/project.vue]: It was kept for future use but can be removed.

File: src/pages/project-search/project-search.vue (hunk 4 above)
Comment (ms10398) [reply src/pages/project-search/project-search.vue]: This line isnt added or edited by me.

So should I change it?

Comment (gregory-latinier) [reply src/pages/project-search/project-search.vue]: Yes please. Try to clean the code when you work on a file.

**Pull request merged**